core/arch/arm/kernel/static_ta.c

```diff
@@ -0,0 +1,186 @@
+			continue;
+		}
+	}
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result static_ta_enter_open_session(struct tee_ta_session *s,
+			struct tee_ta_param *param, TEE_ErrorOrigin *eo)
+{
+	TEE_Result res;
+	struct static_ta_ctx *stc = to_static_ta_ctx(s->ctx);
+
+	tee_ta_set_current_session(s);
+	res = tee_ta_param_pa2va(s, param);
+	if (res != TEE_SUCCESS) {
+		*eo = TEE_ORIGIN_TEE;
+		return res;
+	}
+
+	*eo = TEE_ORIGIN_TRUSTED_APP;
+	if (s->ctx->ref_count == 1) {
+		res = stc->static_ta->create_entry_point();
+		if (res != TEE_SUCCESS)
+			goto out;
+	}
+	res = stc->static_ta->open_session_entry_point(param->types,
+					param->params, &s->user_ctx);
+
+out:
+	tee_ta_set_current_session(NULL);
+	return res;
+}
+
+static TEE_Result static_ta_enter_invoke_cmd(struct tee_ta_session *s,
+			uint32_t cmd, struct tee_ta_param *param,
+			TEE_ErrorOrigin *eo)
+{
+	TEE_Result res;
+	struct static_ta_ctx *stc = to_static_ta_ctx(s->ctx);
+
+	tee_ta_set_current_session(s);
+	res = tee_ta_param_pa2va(s, param);
+	if (res != TEE_SUCCESS) {
+		*eo = TEE_ORIGIN_TEE;
+		return res;
+	}
+
+	*eo = TEE_ORIGIN_TRUSTED_APP;
+	res = stc->static_ta->invoke_command_entry_point(s->user_ctx, cmd,
+					param->types, param->params);
+	tee_ta_set_current_session(NULL);
+	return res;
+}
+
+static void static_ta_enter_close_session(struct tee_ta_session *s)
+{
+	struct static_ta_ctx *stc = to_static_ta_ctx(s->ctx);
+
+	tee_ta_set_current_session(s);
+	stc->static_ta->close_session_entry_point(s->user_ctx);
+	if (s->ctx->ref_count == 1)
+		stc->static_ta->destroy_entry_point();
+	tee_ta_set_current_session(NULL);
+}
+
+static void static_ta_destroy(struct tee_ta_ctx *ctx __unused)
+{
+	/* Nothing to do */
+}
+
+static const struct tee_ta_ops static_ta_ops = {
+	.enter_open_session = static_ta_enter_open_session,
+	.enter_invoke_cmd = static_ta_enter_invoke_cmd,
+	.enter_close_session = static_ta_enter_close_session,
+	.destroy = static_ta_destroy,
+};
+
+
+/* Defined in link script */
+extern const struct static_ta_head __start_ta_head_section;
+extern const struct static_ta_head __stop_ta_head_section;
+
+/*-----------------------------------------------------------------------------
+ * Initialises a session based on the UUID or ptr to the ta
+ * Returns ptr to the session (ta_session) and a TEE_Result
+ *---------------------------------------------------------------------------*/
+TEE_Result tee_ta_init_static_ta_session(const TEE_UUID *uuid,
+			struct tee_ta_session *s)
+{
+	struct static_ta_ctx *stc = NULL;
+	struct tee_ta_ctx *ctx;
+	const struct static_ta_head *ta;
+
+	DMSG("   Lookup for Static TA %pUl", (void *)uuid);
+
+	ta = &__start_ta_head_section;
+	while (true) {
+		if (ta >= &__stop_ta_head_section)
+			return TEE_ERROR_ITEM_NOT_FOUND;
+		if (memcmp(&ta->uuid, uuid, sizeof(TEE_UUID)) == 0)
+			break;
+		ta++;
+	}
+
+	/* Load a new TA and create a session */
+	DMSG("      Open %s", ta->name);
+	stc = calloc(1, sizeof(struct static_ta_ctx));
+	if (stc == NULL)
```

Comment:
@vchong we shouldn't avoid the check it self, but we should use `if (!stc)` instead of `if (stc == NULL)`. Sorry for my sloppy writing.